lib/pxlogger.js

```diff
@@ -9,20 +9,25 @@ class PxLogger {
         this.appId = '';
     }
 
+    setCustomLogger(customLogger) {
+        this.customLogger = customLogger;
+    }
+
     init(pxConfig) {
         this.debugMode = pxConfig.conf.DEBUG_MODE;
         this.appId = pxConfig.conf.PX_APP_ID;
+        this.customLogger = pxConfig.conf.CUSTOM_LOGGER;
     }
 
     debug(msg) {
         if (this.debugMode && msg) {
-            console.info(`[PerimeterX - DEBUG][${this.appId}] - ${msg}`);
+            (this.customLogger ? this.customLogger : console).info(`[PerimeterX - DEBUG][${this.appId}] - ${msg}`);
```

Comment:
I would remove this logic here and update the init function to `this.logger = pxConfig.conf.CUSTOM_LOGGER || console`.

That way here, you can call `this.logger.info` consistently.

Reply:
lol yeah, I actually had it kinda like that before, but changed it to this because I thought it was weird that [this line in my test](https://github.com/PerimeterX/perimeterx-node-core/pull/138/files/7ddb7a6f7450ca3cef73e2a6858d5543a5b6eace#diff-a38142a3ad162b2ddf277c98ecc13d78850e4d23819bc43a567d1f0434fd908cR117) was _conf.**logger.logger**.should.be.exactly(params.customLogger)_. But yup, changed it back 👍